From 053e1d40f510214343e812a1ae9fbc13b7c8858b Mon Sep 17 00:00:00 2001
From: Yihang Li <liyihang9@h-partners.com>
Date: Thu, 31 Jul 2025 14:43:55 +0800
Subject: [PATCH 5/9] sysSentry: Fix issue inconsistent status and result after
 single inspection

driver inclusion
category: feature
bugzilla: https://gitee.com/src-openeuler/sysSentry/issues/ID1UOY
CVE: NA

----------------------------------------------------------------------

When configured for a single inspection, the inspection status is set to
FAILED after the inspection is completed, but the inspection result is
empty, which does not meet expectations.

All inspection threads should directly release resources, return the
corresponding results, and exit after completion.

Signed-off-by: Yihang Li <liyihang9@h-partners.com>
Signed-off-by: Qizhi Zhang <zhangqizhi3@h-partners.com>
---
 src/sentryPlugins/soc_ring_sentry/tc_ring_one.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/sentryPlugins/soc_ring_sentry/tc_ring_one.c b/src/sentryPlugins/soc_ring_sentry/tc_ring_one.c
index 9473d68..e8740c0 100755
--- a/src/sentryPlugins/soc_ring_sentry/tc_ring_one.c
+++ b/src/sentryPlugins/soc_ring_sentry/tc_ring_one.c
@@ -410,13 +410,14 @@ static int tc_ring_one_exec(struct tc_ring_one_config *config)
         pthread_join(config->tc_core_threads[i], NULL);
     }
 
-    i = i - 1;
     if (config->err_flag > 0) {
         ret = TC_RING_ONE_FAIL;
     } else {
         ret = TC_RING_ONE_SUCCESS;
     }
 
+    goto out;
+
 pthread_bind_fail:
     if (!is_core_invalid(config, i)) {
         pthread_cancel(config->tc_core_threads[i]);
@@ -433,6 +434,7 @@ pthread_create_fail:
         pthread_join(config->tc_core_threads[i], NULL);
     }
 
+out:
     tc_ring_one_release(config);
     free(config->tc_core_threads);
     config->tc_core_threads = NULL;
-- 
2.33.0

